### PR TITLE
fix: fix #263 upsert attributes should use defaultValue while there i…

### LINKED ISSUE #263: [bug] upsert attributes should use defaultValue while there is no value assigned
```js

class User {
  id: number;
  email: string;
  nickname: string;
  status: {
     type: number,
     defaultValue: 1
  };
}

// INSERT INTO `users` ( `email`, `nickname`, `status`) VALUES('sss', 'hello', 1) ****
User.upsert({ email: 'sss', nickname: 'hello' });

```

### PR DESCRIPTION
**Pull request by JimmyDaddy**
…s no value assigned

### CODE REVIEW

File: src/bone.js [hunk 1]
```diff
@@ -846,9 +847,13 @@ class Bone {
     const data = {};
     const Model = this;
     const { attributes } = Model;
-    for (const name in values) {
-      if (this.hasAttribute(name)) {
-        data[name] = values[name];
+    for (const key in attributes) {
+      const attribute = attributes[key];
+      if (attribute.primaryKey) continue;
+      if (!values[key] && attribute.defaultValue != null) {
```
Comment (cyjake) [inline src/bone.js:853]: How about values such as `false`, `0`, or `''`? I think `values[key] == null` should be preferred.

File: test/integration/suite/basics.test.js [hunk 2]
```diff
@@ -912,6 +912,13 @@ describe('=> Basic', () => {
           count = await Tag.count();
           assert.equal(count, 1);
         });
+
+        it('Bone.upsert should assigne defaultValue automatically', async () => {
```
Comment (cyjake) [inline test/integration/suite/basics.test.js:916]: assign

File: test/unit/adapters/sequelize.test.js [hunk 3]
```diff
@@ -1743,11 +1741,22 @@ describe('validator should work', () => {
         status: 1,
       }, { validate: false });
 
+      await new Promise((resolve) => {
+        setTimeout(() => {
+          resolve(true);
+        }, 1000);
+      });
```
Comment (cyjake) [inline test/unit/adapters/sequelize.test.js:1748]: why is this sleep necessary?

File: src/contants.js [hunk 4]
```diff
@@ -8,6 +8,23 @@ const AGGREGATOR_MAP = {
...
+  gmtDeleted: 'deletedAt',
+};
+
+const LEGACY_TIMESTAMP_CLOUMN_MAP = {
+  created_at: 'gmt_create',
+  updated_at: 'gmt_modified',
+  deleted_at: 'gmt_deleted',
+};
+
+const TIMESTAMPS_ATTRNAME = [ 'createdAt', 'updatedAt', 'deletedAt' ];
```
Comment (cyjake) [inline src/contants.js:23]: TIMESTAMP_NAMES

File: src/realm.js [hunk 5]
```diff
@@ -8,6 +8,7 @@ const { findDriver } = require('./drivers');
 const { camelCase } = require('./utils/string');
 const sequelize = require('./adapters/sequelize');
 const Raw = require('./raw');
+const { LEGACY_TIMESTAMP_MAP } = require('./contants');
```
Comment (cyjake) [inline src/realm.js:11]: constants

didn't realize this typo until now :-(

File: src/contants.js [hunk 6]
```diff
@@ -8,6 +8,23 @@ const AGGREGATOR_MAP = {
...
+  gmtCreate: 'createdAt',
+  gmtModified: 'updatedAt',
+  gmtDeleted: 'deletedAt',
+};
+
+const LEGACY_TIMESTAMP_CLOUMN_MAP = {
+  created_at: 'gmt_create',
+  updated_at: 'gmt_modified',
+  deleted_at: 'gmt_deleted',
+};
```
Comment (cyjake) [inline src/contants.js:21]: LEGACY_TIMESTAMP_COLUMN_MAP

File: test/unit/adapters/sequelize.test.js (hunk 3 above)
Comment (JimmyDaddy) [reply test/unit/adapters/sequelize.test.js:1748]: Verifying createdAt will not be updated at `upsert`

File: src/realm.js (hunk 5 above)
Comment (JimmyDaddy) [reply src/realm.js]: > constants
> 
> didn't realize this typo until now :-(

LOL, my left hand always quick than my right hand🙈

**Review by cyjake — APPROVED** [review]

**Pull request merged**